src/item.cpp

```diff
@@ -94,6 +94,24 @@ item::item(const std::string new_type, unsigned int turn, bool rand, const hande
     // TODO: some item types use the same member (e.g. charges) for different things. Handle or forbid this.
     if( type->gun ) {
         charges = 0;
+        if( type->gun->built_in_mods.size() > 0 ){
+            for( unsigned int i = 0; i < type->gun->built_in_mods.size(); i++ ){
```

Comment:
You want to use range-based for loops here. This also get rid of the call to `at`, which can throw, but the exception is nowhere caught.

```C++
for( auto &gm : type->gun->build_in_mods ) {
    item temp( gm, turn );
    ...
```